src/assets/scripts/client/airport/runway/RunwayCollection.js

```diff
@@ -0,0 +1,264 @@
+import _find from 'lodash/find';
+import _forEach from 'lodash/forEach';
+import _isArray from 'lodash/isArray';
+import BaseCollection from '../../base/BaseCollection';
+import RunwayModel from './RunwayModel';
+import RunwayRelationshipModel from './RunwayRelationshipModel';
+import { extrapolate_range_clamp } from '../../math/core';
+import { degreesToRadians } from '../../utilities/unitConverters';
+
+/**
+ * Collection of `RunwayModel`s
+ *
+ * Provides methods to reason about the various `RunwayModel`s belonging to an airport
+ *
+ * @class RunwayCollection
+ * @extends BaseCollection
+ */
+export default class RunwayCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @param runwayJson {array<object>}
+     * @param airportPositionModel {StaticPositionModel}
+     */
+    constructor(runwayJson, airportPositionModel) {
+        super();
+
+        if (!_isArray(runwayJson)) {
+            throw new TypeError(
+                `Invalid parameter passed to RunwayCollection. Expected an array but found ${typeof runwayJson}`
+            );
+        }
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property _items
+         * @type {array<RunwayModel>}
+         * @default []
+         */
+
+         /**
+          * @inherited
+          * @memberof BaseCollection
+          * @property length
+          * @type {number}
+          * @default #_items.length
+          */
+
+        /**
+         * @property _airportPositionModel
+         * @type {StaticPositionModel}
+         * @default null
+         * @private
+         */
+        this._airportPositionModel = null;
```

Comment:
Right, that whole "reference position" thing. Ugh. That's another thing to rework eventually. I'll add an issue for it now so it can get in the queue somewhere down the line.